### ISSUE TITLE
The run-ahead function doesn't seem to work properly sometimes

### ISSUE THREAD

**Issue by guljam** [libretro/RetroArch#15381]
## Description

Mame core 0.255 
video driver : glcore
hard gpu sync on, sync frame 1
shader : newpixie-crt.slangp
3 frame run-ahead
automatic frame delay ON

https://github.com/libretro/RetroArch/assets/15816034/9e17e270-46f5-4b4a-b10b-d98f93a4ed05

Can the run-ahead function cause a bug depending on the state saving function depending on the core? I'm not sure what the cause is, but I think you need to check the run-ahead function of retroarch itself.

### Expected behavior

I'm playing Mr. Driller 2 on Namco system 10
As shown in the attached video, if you eat an AIR item, the item moves to the right. Oxygen is increased by 20%

### Actual behavior

After acquiring the Oxygen item, the Oxygen moves to the right, but suddenly teleports back to its original location, preventing it from being acquired.

### Steps to reproduce the bug

This happens very intermittently

### Bisect Results

[Try to bisect and tell us when this started happening]

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.15.0 git version cc1bc25

### Environment information

- OS: windows 11 22H2 Build 22612.1702
- Compiler:

**Comment by hizzlekizzle** [comment]
I don't think many MAME drivers' savestates have the determinism required to avoid all runahead weirdness. FBNeo typically makes sure their drivers are up to the task, but obviously it supports far fewer machines.

**Comment by guljam** [comment]
@hizzlekizzle 

This seems to be a problem with my setup, closed the issue right away but thanks for the reply.

Although I solved the problem by decreasing the run-ahead by 1 frame. A slight delay is felt.

"Mr. Driller" game is a game where input lag is important like a rhythm game.

MAME's Namco system 10 emulation is great, but even retroarch's features don't seem to let me play with the feel of the original hardware.

Although the version ported to PC can be played without input lag. It doesn't have the old arcade feel, and they're all shit. haha

**Comment by guljam** [comment]
@hizzlekizzle 

Thank you for answer. I immediately closed this as I don't think it has anything to do with the Retroarch Issue.

I love Mr. Driller games, I've played them on real hardware and the run-ahead 3 frames gave me a very close feel to real hardware. But sometimes due to the previous frame jump bug I stopped playing now. (With 2 frames run-ahead there is no bug, but the slight lag bothers me haha)

The Mr. Driller 1 game using the Namco System12 driver did not jump to the previous frame at all even when played with the same 3 frame delay.

My guess is that maybe the Namco System 10 driver was written recently so there is a problem with the state save feature.

This can't be fixed in Retroarch, right?

I'm not sure if this is a driver problem or a problem with the game itself, but hopefully it will be resolved someday.
